### ISSUE TITLE
Card due date notification works only once

### ISSUE THREAD

**Issue by Choukajohn** [nextcloud/deck#2239]
**Describe the bug**
The notification when a card reach the due date is working only once.
After the first notification, if I set a new due date on the same card, notification is not working anymore.

**To Reproduce**
1. Create a new card
2. Add a due date and hour
3. Wait for the notification
4. Once due date is reached, modify the due hour and save the card.
5. Notification is not working for the new due date / hour

If a new card is created, notification is working for the new card but only once.

**Expected behavior**
Notifications should work each time the due date and hour are modified

<details>
<summary>Server details</summary>

**Operating system**: Ubuntu 19.10 kernel 5.3.0-64

**Web server:** Apache2

**Database:** MySQL 8.0.20

**PHP version:** 7.3.11

**Nextcloud version:** 19.0.1 with cron service activated

**Where did you install Nextcloud from:** GitHub

**Signing status:** No errors have been found.

**List of activated apps:**
- accessibility: 1.5.0
  - activity: 2.12.0
  - admin_audit: 1.9.0
  - bruteforcesettings: 1.6.0
  - cloud_federation_api: 1.2.0
  - comments: 1.9.0
  - contactsinteraction: 1.0.0
  - dav: 1.15.0
  - deck: 1.0.5
  - documentserver_community: 0.1.7
  - federatedfilesharing: 1.9.0
  - federation: 1.9.0
  - files: 1.14.0
  - files_external: 1.10.0
  - files_pdfviewer: 1.8.0
  - files_rightclick: 0.16.0
  - files_sharing: 1.11.0
  - files_trashbin: 1.9.0
  - files_versions: 1.12.0
  - files_videoplayer: 1.8.0
  - firstrunwizard: 2.8.0
  - logreader: 2.4.0
  - lookup_server_connector: 1.7.0
  - news: 14.1.11
  - nextcloud_announcements: 1.8.0
  - notes: 3.6.4
  - notifications: 2.7.0
  - oauth2: 1.7.0
  - onlyoffice: 4.3.0
  - password_policy: 1.9.1
  - photos: 1.1.0
  - privacy: 1.3.0
  - provisioning_api: 1.9.0
  - recommendations: 0.7.0
  - serverinfo: 1.9.0
  - settings: 1.1.0
  - sharebymail: 1.9.0
  - support: 1.2.1
  - survey_client: 1.7.0
  - systemtags: 1.9.0
  - tasks: 0.13.3
  - text: 3.0.1
  - theming: 1.10.0
  - twofactor_backupcodes: 1.8.0
  - updatenotification: 1.9.0
  - viewer: 1.3.0
  - workflowengine: 2.1.0

</details>

**Comment by jakobroehrl** [comment]
can reproduce it. 

Check the code here:
https://github.com/nextcloud/deck/blob/7e183d6e99a4d4a1d9cb7a8e7bab23800c4b398c/lib/Db/CardMapper.php#L86
https://github.com/nextcloud/deck/blob/52febb396cc79fd162efdd87550947f35d3e3f84/lib/Notification/NotificationHelper.php#L75-L99

**Comment by juliushaertl** [comment]
Pushed a fix for this to #2430 in 87a7097
